src/mds/Mutation.h

```diff
@@ -293,24 +293,30 @@ struct MDRequestImpl : public MutationImpl {
     // keep these default values synced to MutationImpl's
     Params() : attempt(0), client_req(NULL),
         triggering_slave_req(NULL), slave_to(MDS_RANK_NONE), internal_op(-1) {}
+    const utime_t& get_recv_stamp() const {
+      return initiated;
+    }
+    const utime_t& get_throttle_stamp() const {
+      return throttled;
+    }
+    const utime_t& get_recv_complete_stamp() const {
+      return all_read;
+    }
+    const utime_t& get_dispatch_stamp() const {
+      return dispatched;
+    }
   };
-  MDRequestImpl(const Params& params, OpTracker *tracker) :
-    MutationImpl(tracker, params.initiated,
-		 params.reqid, params.attempt, params.slave_to),
+  MDRequestImpl(const Params* params, OpTracker *tracker) :
```

Comment:
Why are you changing this to a const pointer? It was previously a const reference because that's our style guide and makes very clear there won't be any changes or shallow copies of the parameter.

Reply:
@gregsfortytwo 
Hi, thanks for the review:-)
I changed this to a pointer because, in create_request, I can only reference the get_XXX_stamp() method through either "->" or "." and, in OSD and monitor, the parameter of the create_request method is a pointer, so I choose to change the "params" of MDRequestImpl to a pointer.

Reply:
Oh of course, that makes sense,